app/exerlog/lib/src/feature/authentication/view/landing_screen.dart

```diff
@@ -0,0 +1,171 @@
+import 'package:exerlog/UI/global.dart';
+import 'package:exerlog/UI/workout/workout_page.dart';
+import 'package:exerlog/src/core/base/base_state.dart';
+import 'package:exerlog/src/core/base/extensions/context_extension.dart';
+import 'package:exerlog/src/feature/authentication/controller/authentication_controller.dart';
+import 'package:exerlog/src/feature/authentication/widgets/google_signin_button.dart';
+import 'package:exerlog/src/feature/authentication/widgets/login_form.dart';
+import 'package:exerlog/src/feature/authentication/widgets/signup_form.dart';
+import 'package:exerlog/src/widgets/gradient_button.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class LandingScreen extends ConsumerStatefulWidget {
+  const LandingScreen({Key? key}) : super(key: key);
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _LandingScreenState();
+}
+
+class _LandingScreenState extends ConsumerState<LandingScreen>
+    with SingleTickerProviderStateMixin {
+  /// Form Keys
+  final _loginFormKey = GlobalKey<FormState>();
+  final _signUpFormKey = GlobalKey<FormState>();
+
+  /// Tab
+  late TabController _tabController;
+  int _tabIndex = 0;
+  List<Tab> tabs = <Tab>[
+    Tab(
+      child: Text(
+        "Login",
+
+        /// TODO: Handle text style properly (applies everywhere)
+        style: loginOptionTextStyle,
+      ),
+    ),
+    Tab(
+      child: Text(
+        "Sign up",
+        style: loginOptionTextStyle,
+      ),
+    ),
+  ];
+
+  void _tabSwitchListener(int value) {
+    setState(() {
+      _tabIndex = value;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(length: tabs.length, vsync: this);
+    _tabController.addListener(() {
+      _tabSwitchListener(_tabController.index);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final _controller = ref.read(AuthenticationController.controller);
+
+    ref.listen(AuthenticationController.provider, (_, state) {
+      if (state is SuccessState) {
+        Navigator.of(context).pushReplacement(

```

Comment:
Not sure if in the scope of this PR, but we should probably use named routes, this way we get and overview of all the routes the app has. Also I prefer putting all the navigation actions at the bottom of the widget outside of the build function. This way we get a quick overview of all the places a widget can navigate to.

Reply:
Wondering if we should tackle this in this PR or leave it for an app wide refactoring of routings?

Reply:
Also, want to simplify the navigation boilerplate. May be utilising the context extension 🤔

Reply:
Hmm yeah maybe we should do it in another PR seeing as this one is already pretty big. As for the navigation I was thinking something like this, do you have something else in mind?:

```
class Widget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return  //widget
  }

  void goToDetail() => Navigator.of(context).pushNamed(RouteNames.detail);

  void goToOVerview() => Navigator.of(context).pushNamed(RouteNames.overview);
}
```

Reply:
Or maybe something like this would be bette: 

```
  void goToDetail() => Navigator.of(context).pushNamed(DetailPage.routeName);

  void goToOVerview() => Navigator.of(context).pushNamed(Overviewpage.routeName);
```

and then define a status variable in the pageWidget that references the right route from RouteNames

Reply:
I prefer the one with `RouteNames.<screen>`.
But yeah this is what I am thinking. I just want to minimise it down to 
`context.PushNamed(RouteNames.detail);`

Reply:
That would indeed be more clean, my only fear is that we'd be putting to much navigation logic inside the context extension, but I think it's definitely worth at try.

Reply:
What if we extract those to a different extension? Then navigation logic will be separated from other context extension. Or we can also think of a wrapper class, to hide the boilerplate.

```
Class Navigation {
    /// Handle navigation logic here.
}
```

Reply:
Or maybe we could write a mixin on the state this way we'd get access to the context and just add methods to this mixin that navigate to the different pages.

```
abstract class Navigation {
  void goToDetail();

  void goToOverview();
}

mixin NavigationMixin<T extends StatefulWidget> on State<T> implements Navigation {
  NavigatorState get navigator => Navigator.of(context);

  @override
  void goToDetail() => navigator.pushNamed(RouteNames.detail);

  @override
  void goToOverview() => navigator.pushNamed(RouteNames.overview);
}
```

And then we could use it in a widget like this: 


```
class Widget extends StatelessWidget with NavigationMixin {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(onTap:goToDetail )
  }
```

Reply:
This is nice. 
But I am thinking about simplified solutions. The idea of declaring methods for every possible navigation and calling method might get out of hand.

Instead a way to just say `navigate(Routes.destination)` or `context.pushNamed(Routes.destination)` sounds lot easier to maintain. Because once possible all navigation logic is implemented, we don't need to modify it any more. We can just keep adding routes in the Route list and just navigate.

But in the above mentioned process, one has to add 
1. Route Name
2. Add method
3. Add implementation
4. use Mixin where they need it
5. Then remember what method was declared for the route they want to go.

But what I am thinking is: (let's say we use context extension)
1. Just add the route name pointing to the correct widget
2. use context.pushNamed(Routes.name)

No need to remember new navigation terminology, because in context extension we will stick to the ones given by the framework.

Reply:
@jorre127 I am resolving this for now. But let's discuss this over an issue referencing this conversation thread. Let's get this changes merged. Lot's of other PRs are depending on this.

Reply:
The context extension indeed seems like a good idea then, as long as its separated from the other context extensions.